iroh-base/src/node_addr.rs

```diff
@@ -74,13 +74,13 @@ impl NodeAddr {
     pub fn from_parts(
         node_id: PublicKey,
         relay_url: Option<RelayUrl>,
-        direct_addresses: Vec<SocketAddr>,
+        direct_addresses: BTreeSet<SocketAddr>,
```

Comment:
Could we receive iterators instead? All calls already are performed over iterators. 
```suggestion
        direct_addresses: impl IntoIterator<Item = SocketAddr>,
```
along with a call to `BTreeSet::from_iter`

With this most calls become a bit cleaner.
For example:
```diff
-            node: NodeAddr::from_parts(peer, relay_url, BTreeSet::from_iter([addr])),
+            node: NodeAddr::from_parts(peer, relay_url, [addr]),
```

Reply:
Yes, this works out well and is a much nicer API.  Great idea!